### PR TITLE
feat(routemonitor): make code work with tests

### PR DESCRIPTION
**Pull request by georgettica**
Rewrote all of the code again with tests in mind, so now all of the code
is seperated

- code is now formatted with BDD in mind
- suite_test is removed for requiring etcd to run
- add to .gitignore vim tmp files

### CODE REVIEW

**Comment by RiRa12621** [comment]
/assign @NautiluX 
Manuel loves to review some nice tests <3

File: .gitignore
Comment (RiRa12621) [inline .gitignore:21]: this should be in your global gitignore

File: config/rbac/role.yaml
Comment (RiRa12621) [inline config/rbac/role.yaml:42]: are we sure we never need to update servicemonitors?

File: config/rbac/role.yaml
Comment (RiRa12621) [inline config/rbac/role.yaml:30]: what if the image changes and we need to update the deployment

File: main.go
Comment (RiRa12621) [inline main.go:96]: what's this one about?

File: controllers/routemonitor/routemonitor_controller.go
Comment (RiRa12621) [inline controllers/routemonitor/routemonitor_controller.go:22]: i personally prefer logrus but no hard feelings

**Review by RiRa12621 — APPROVED** [review]
Awesome work overall, just some tiny comments

**Comment by RiRa12621** [comment]
/hold
some time for you to go through the comments and someone else to give eyes.

File: config/rbac/role.yaml
Comment (georgettica) [reply config/rbac/role.yaml:42]: not the spec section, status yes

File: config/rbac/role.yaml
Comment (georgettica) [reply config/rbac/role.yaml:30]: then we update the role and update the deployment

currently if the deployment exist I leave it alone. are there more tests we would like to do on the deployment?

File: main.go
Comment (georgettica) [reply main.go:96]: when I tried to add the `CreateBlackBoxExporter` there, it didn't run, as `mgr.Start` is blocking

File: controllers/routemonitor/routemonitor_controller.go
Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller.go:22]: I didn't decide that. that is the logger of operator-sdk

File: main.go
Comment (RiRa12621) [reply main.go:96]: maybe we make the error message a tiny bit more meaningful to reflect that ;)

Comment (georgettica) [reply main.go:96]: it's a log message, will update with this fix. thanks!

File: controllers/routemonitor/routemonitor_controller.go
Comment (NautiluX) [inline controllers/routemonitor/routemonitor_controller.go:51]: as said before, I think this should always happen when the first thing is deployed, and get removed if it's not needed anymore.

File: controllers/routemonitor/routemonitor_controller.go
Comment (NautiluX) [inline controllers/routemonitor/routemonitor_controller.go:77]: This code reads like a novel. Love it.

File: main.go
Comment (georgettica) [reply main.go]: rewrote, can you review the new message and see if it's better?

**Comment by georgettica** [comment]
thanks for the label @RiRa12621 ! I squashed and removed the label

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (NautiluX) [inline controllers/routemonitor/routemonitor_controller_supplement.go:179]: *Exporter
missing in function names

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (NautiluX) [inline controllers/routemonitor/routemonitor_controller_supplement.go:275]: I don't understand this TODO, this port is set at the service as well, isn't it?

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (NautiluX) [inline controllers/routemonitor/routemonitor_controller_supplement.go:217]: Do we need this? If not, remove

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (NautiluX) [inline controllers/routemonitor/routemonitor_controller_supplement.go:241]: Do we need this? If not, remove

File: .gitignore
Comment (NautiluX) [reply .gitignore:21]: I'd say keep it.

**Review by NautiluX — COMMENTED** [review]
Looks good, see comments. The when should a balckboxexporter be deployed or removed question can be answered in a separate PR.

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (RiRa12621) [reply controllers/routemonitor/routemonitor_controller_supplement.go:275]: I think that's an artifact from the code i initially hacked together

File: controllers/routemonitor/routemonitor_controller.go
Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller.go:51]: not sure where to put it if it only happens once. tried to put in main.go but wasn't executed / was too early

File: controllers/routemonitor/routemonitor_controller.go
Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller.go:77]: ❤️ ❤️

**Review by georgettica — COMMENTED** [review]
added commits. will squash when you feel that it's good

**Comment by NautiluX** [comment]
I think you can remove the comments and we add it later if we think it's necessary. But I'll leave that up to you, remove hold whenever you're ready
/approve
/lgtm

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (RiRa12621) [inline controllers/routemonitor/routemonitor_controller_supplement.go:205]: we need to move that to quay

Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller_supplement.go:205]: like, to a local quay or a pipelined thing with quay?

Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller_supplement.go:205]: cuz I can push it to my repo for now but it will not have pipelines on it

File: controllers/routemonitor/routemonitor_controller.go
Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller.go]: @NautiluX

File: controllers/routemonitor/routemonitor_controller_supplement.go
Comment (georgettica) [reply controllers/routemonitor/routemonitor_controller_supplement.go:217]: @NautiluX

**Comment by georgettica** [comment]
ok @NautiluX will do now

**Comment by georgettica** [comment]
removed comments and will merge
/lgtm
/approve

**Comment by georgettica** [comment]
@RiRa12621 @NautiluX  can you please re`/lgtm` for pr? squashed and merged

**Comment by georgettica** [comment]
thanks!

**Pull request merged**
